core/matchdata/byteoffset_spec.rb

```diff
@@ -0,0 +1,65 @@
+require_relative '../../spec_helper'
+
+describe "MatchData#byteoffset" do
+  ruby_version_is "3.2" do
+    it "returns beginning and ending byte-based offset of n-th match" do
+      m = /(.)(.)(\d+)(\d)/.match("THX1138.")
+
+      m.byteoffset(0).should == [1, 7]
+      m.byteoffset(2).should == [2, 3]
+      m.byteoffset(4).should == [6, 7]
+    end
```

Comment:
**suggestion:** It makes sense to mention that 0 element is the whole matched substring, and all the subsequent elements (1st, 2d, etc) are capturing groups